src/main/java/org/jabref/logic/formatter/bibtexfields/RemoveNewlinesFormatter.java

```diff
@@ -24,7 +24,7 @@ public String getKey() {
     public String format(String value) {
         Objects.requireNonNull(value);
 
-        return value.replace("\r\n", " ").replace("\n", " ").trim();
+        return value.replace("\r\n", " ").replace("\n", " ").replace("\r", " ").trim();
```

Comment:
I would also add a special treatment for hyphenated words.
Algorithm sketch: If the last character before the line termination symbol is a minus, replace the minus and the line-terminator with an empty string.

```
"...Jab-\n"
"ref is cool!\n" 
```
=>
```
"...JabRef is cool!"
```

Reply:
What about Key-Account-Manager?

Reply:
Well... bad luck? words with a dash are quite rare, at least less frequent than hyphens at line breaks.

This is a problem, where we cant make a good decision.

Reply:
Words with a dash are quite rare in English, maybe...

Reply:
We can just add your solution and see if someone complains. I guess leaving the chars as they are is also not a bad decision.

Reply:
We've discussed this in the DevCall: Please go for the solution suggested by @LinusDietz